src/html/fetch-markdown.js

```diff
@@ -86,24 +81,19 @@ async function fetch(
     timeout,
     time: true,
   };
+
   logger.debug(`fetching Markdown from ${options.uri}`);
   try {
-    const response = await client(options);
-    return {
-      content: {
-        body: response,
-        sources: [...sources, options.uri],
-      },
-    };
+    content.body = await client(options);
+    setdefault(content, 'sources', []).push(options.uri);
   } catch (e) {
-    if (e && e.statusCode && e.statusCode === 404) {
-      return bail(logger, `Could not find Markdown at ${options.uri}`, e, 404);
-    } else if ((e && e.response && e.response.elapsedTime && e.response.elapsedTime > timeout) || (e && e.cause && e.cause.code && (e.cause.code === 'ESOCKETTIMEDOUT' || e.cause.code === 'ETIMEDOUT'))) {
```

Comment:
It's no longer really necessary…
The pipeline now catches any errors during pipeline execution, sends an error message without specific information to the user and prints the error plus additional debug information to the log.
This is a significant improvement as the old bail() way would not print the stack trace of the error or any information regarding which pipeline step we are in.

The only drawback of the change is that a 500 is passed to the user instead of a 504, but I personally would argue that there is no real reason to do that anyways: Only errors which indicate a user error (e.g. type on the url leads to 404) should be communicated to the user; exposing any other errors just communicates  potentially private information to third parties.

A gateway timeout is nothing a user can do something about, so we give them no information while the developer should receive any relevant information from the printed error.

Reply:
I didn't mean that the timeout should be returned to the user. but the nicer error message in the log is also removed. i.e. I would keep the `if()` and the just throw the timeout error.

Reply:
Why? The default error message is quite sufficient to debug what's going on? The user knows the url and any options anyway from the default error logging, so the nicer error message doesn't really give any additional information…

Reply:
By the way, I did add an extra logging for any errors giving the url and the options, so that should cover it :)

Reply:
ok. coo!

Reply:
I reverted the error handling to the previous state. @koraa we can discuss this again in a separate issue. changing the pipeline behaviour is enough for one PR; let's keep the semantics of `fetch-markdown()` consistent.